AlgebraInLean/World0LeanIntroSOLUTIONS/Sheet2Solutions.lean

```diff
@@ -0,0 +1,269 @@
+/-
+This is a solutions sheet.
+-/
+
+import Mathlib.Tactic
+
+/-Credit for some exercises goes to Kevin Buzzard and his Formalizing Mathematics
+course OR to the Natural Number Game. If you would like to learn more about Lean,
+Buzzard's course goes more in depth in relation to numerous undergraduate math topics.
+When exercises are from either of these sources, they will be clearly marked so as
+to give credit.-/
+
+/-
+Many basic tactics in Lean are best introduced through logic exercises.
+You already know rfl and rw from the Natural Number Game (NNG), and you are also
+already familiar with a few other basic tactics. However, keep in mind that the
+way these tactics work in the Natural Number Game may be slightly different than
+the way that they are actually used in Lean. These differences mostly boil down
+to slight discrepancies in syntax.
+
+Let's start with a few tactics that you are already familiar with from the NNG:
+
+exact
+apply
+intro
+
+To continue getting used to reading Lean and working with the Lean Infoview,
+let's do a few levels from the NNG. Don't forget that you can also use rw and rfl!
+Delete the sorrys and fill in the proofs.
+-/
+
+/-NNG Implication World Level 1-/
+example (x y z : Nat) (h1 : x + y = 37) (h2 : 3 * x + z = 42) : x + y = 37 := by
+  exact h1
+  done
+
+/-NNG Implication World Level 2-/
+/-hint: zero_add still works outside of the NNG-/
```

Comment:
I think you should mention Mathlib and provide a brief description of what it is at the start of this sheet. On line 38, you can mention that zero_add is a theorem proven in mathlib

Reply:
Done!